src/draw.h

```diff
@@ -40,6 +36,27 @@ enum corner_pos {
         _C_LAST = 1 << 5,
 };
 
+struct color {
+        double r;
+        double g;
+        double b;
+        double a;
+};
+
+#define COLOR_UNINIT { -1, -1, -1, -1 }
+#define COLOR_VALID(c) ((c).r >= 0 && (c).g >= 0 && (c).b >= 0 && (c).a >= 0)
+#define COLOR_SAME(c1, c2) ((c1).r == (c2).r && (c1).g == (c2).g && (c1).b == (c2).b && (c1).a == (c2).a)
```

Comment:
This one is just used in tests. I think, it should be moved to `test/helpers.h`.